src/java/org/apache/cassandra/service/accord/txn/TxnReference.java

```diff
@@ -351,4 +407,100 @@ public long serializedSize(TxnReference reference, TableMetadatas tables, Versio
             return size;
         }
     };
+
+    private static class ClusteringColumnData extends AbstractCell<ByteBuffer>
```

Comment:
this is kinda a hack... it was this or refactor the code more.  I could also special case clustering, but felt that this was best as it kept the logic simpler